lightning/src/ln/channelmanager.rs

```diff
@@ -495,7 +497,18 @@ struct PendingInboundPayment {
 /// concrete type of the KeysManager.
 ///
 /// (C-not exported) as Arcs don't make sense in bindings
-pub type SimpleArcChannelManager<M, T, F, R, L> = ChannelManager<Arc<M>, Arc<T>, Arc<KeysManager>, Arc<F>, Arc<R>, Arc<L>>;
+pub type SimpleArcChannelManager<M, T, F, L> = ChannelManager<
+	Arc<M>,
+	Arc<T>,
+	Arc<KeysManager>,
+	Arc<F>,
+	Arc<DefaultRouter<
+		Arc<NetworkGraph<Arc<L>>>,
+		Arc<L>,
+		Arc<Mutex<ProbabilisticScorer<Arc<NetworkGraph<Arc<L>>>, Arc<L>>>>
+	>>,
```

Comment:
I don't see an issue with parameterizing by `ChainMonitor` and taking a `Persist` in `Simple*ChanMan` instead. If a good candidate for parameterization exists, seems like we *should* use it.

Reply:
Hmm... I think the idea for this type alias is to simply using `Arc`s not necessarily defining reasonable defaults. Then again, it's already parameterized with a `KeysManager`. 🤷‍♂️ At very least the docs should be updated accordingly. Just seems any such alias should be defined at a higher level if we are going to be opinionated. Or have another alias like `DefaultArcChannelManager` or the like.

Reply:
I guess the docs aren't super clear. It says "Defining these type aliases prevents issues such as overly long function definitions." And then goes on to state that `KeysManager` was chosen as a concrete type. But we don't do the same with `ChainMonitor`. We can have define the aliases for whatever reason we want, of course. Just looking for some consistency. Changing it may break users, though, if they are relying on it for their own parameterization, which is probably a low probability and can be easily fixed with their own alias.

Reply:
Updated the docs. Browsing the `git blame`, I think the `ChainMonitor` not being specified may be a legacy thing and could be updated for consistency (probably not in this PR though)